config/env/default.js

```diff
@@ -28,5 +28,19 @@ module.exports = {
   sessionKey: 'sessionId',
   sessionCollection: 'sessions',
   logo: 'modules/core/client/img/brand/logo.png',
-  favicon: 'modules/core/client/img/brand/favicon.ico'
+  favicon: 'modules/core/client/img/brand/favicon.ico',
+  uploads: {
+    profileUpload: {
+      dest: './modules/users/client/img/profile/uploads/', //Profile upload destination path
+      limits: {
+        fileSize: 1*1024*1024 //Max file size in bytes
+      },
+      fileFilter: function (req, file, cb) {
+        if (file.mimetype !== 'image/png' && file.mimetype !== 'image/jpg' && file.mimetype !== 'image/jpeg' && file.mimetype !== 'image/gif') {
+            return cb(new Error('Only image files are allowed!'), false);
+        }
+        cb(null, true);
+      }
+    }
```

Comment:
@mleanos I added this filter in the backend because the previous version allowed to upload any type of file using a POST in  /api/users/picture.
Also, ```filefilter``` is an [option](https://github.com/expressjs/multer#multeropts) of multer and I believe that this code can't be used on Angular.

@lirantal We don't force to use ```filefilter``` everywhere, because after these changes, multer will operate in a [independent way on the controller](https://github.com/gustavodemari/mean/commit/736efbf9ab563c4e2d1941ebb6d134322570e98d#diff-028054ee3eec2278e53157ccf4668fe4R59).

@simison Any suggestion where to put?

Some alternatives (feel free to give your opinion or to add suggestions):

Alternative 1:
Remove fileFilter function from config/env/default.js and define in the controller

config/env/default.js
```javascript
uploads: {
    profileUpload: {
      dest: './modules/users/client/img/profile/uploads/', // Profile upload destination path
      limits: {
        fileSize: 1*1024*1024 // Max file size in bytes (1 MB)
      }
    }
  }
```
modules/users/server/controllers/users/users.profile.server.controller.js
```javascript
var upload = multer(config.uploads.profileUpload).single('newProfilePicture');
upload.fileFilter = function (req, file, cb) {
  if (file.mimetype !== 'image/png' && file.mimetype !== 'image/jpg' && file.mimetype !== 'image/jpeg' && file.mimetype !== 'image/gif') {
        return cb(new Error('Only image files are allowed!'), false);
  }
  cb(null, true);
};
```
---
Alternative 2:
Remove fileFilter function from config/env/default.js, create config/lib/multer.js, define a profileUploadFilter in config/lib/multer.js and import config/lib/multer.js on user profile controller

 config/env/default.js
```javascript
uploads: {
    profileUpload: {
      dest: './modules/users/client/img/profile/uploads/', // Profile upload destination path
      limits: {
        fileSize: 1*1024*1024 // Max file size in bytes (1 MB)
      }
    }
  }
```

config/lib/multer.js

```javascript
module.exports.profileUploadFileFilter = function (req, file, cb) {
    if (file.mimetype !== 'image/png' && file.mimetype !== 'image/jpg' && file.mimetype !== 'image/jpeg' && file.mimetype !== 'image/gif') {
        return cb(new Error('Only image files are allowed!'), false);
    }
    cb(null, true);
  };
```

modules/users/server/controllers/users/users.profile.server.controller.js
```javascript
var profileUploadFileFilter = require(path.resolve('./config/lib/multer')).profileUploadFileFilter;
...
var upload = multer(config.uploads.profileUpload).single('newProfilePicture');
upload.fileFilter = profileUploadFileFilter;
```